main.py

```diff
@@ -304,39 +299,56 @@ def get_board(self, access_token_in):
                     }
                 )
             )
-            ws.recv()
-            while True:
-                temp = json.loads(ws.recv())
-                if temp["type"] == "data":
-                    msg = temp["payload"]["data"]["subscribe"]
-                    if msg["data"]["__typename"] == "FullFrameMessageData":
-                        if not temp["id"] in already_added:
-                            imgs.append(
-                                Image.open(
-                                    BytesIO(
-                                        requests.get(
-                                            msg["data"]["name"],
-                                            stream=True,
-                                            proxies=self.GetRandomProxy(),
-                                        ).content
-                                    )
+
+
+        imgs = []       
+        logger.debug("A total of {} canvas sockets opened", len(canvas_sockets))
+        while len(canvas_sockets) > 0:
+            temp = json.loads(ws.recv())
+            logger.debug("Waiting for WebSocket message")
+            if temp["type"] == "data":
+                logger.debug("Received WebSocket data type message")
+                msg = temp["payload"]["data"]["subscribe"]
+                if msg["data"]["__typename"] == "FullFrameMessageData":
+                    logger.debug("Received full frame message")
+                    img_id = int(temp["id"])
+                    logger.debug("Image ID: {}", img_id)
+                    if img_id in canvas_sockets:
+                        logger.info("Getting image: {}", msg["data"]["name"])
+                        imgs.append(
+                            Image.open(
+                                BytesIO(
+                                    requests.get(
+                                        msg["data"]["name"], stream=True
+                                    ).content
                                 )
                             )
-                            already_added.append(temp["id"])
-                        break
+                        )
+                        canvas_sockets.remove(img_id)
+                        logger.debug("Canvas sockets remaining: {}", len(canvas_sockets))
+
+        for i in range(0, canvas_count-1):
             ws.send(json.dumps({"id": str(2 + i), "type": "stop"}))
 
         ws.close()
 
-        new_img = Image.new("RGB", (1000 * 2, 1000))
-
-        x_offset = 0
-        for img in imgs:
-            new_img.paste(img, (x_offset, 0))
-            x_offset += img.size[0]
-
-        logger.info("Got image.")
-        # new_img.show()
+        #TODO: Multiply by canvas_details["canvasConfigurations"][i]["dx"] and canvas_details["canvasConfigurations"][i]["dy"] instead of hardcoding it
```

Comment:
Dead code

Reply:
This is a reminder to improve the multiplication logic below when/if the canvas grows.

Reply:
Ok for the todo